command/server.go

```diff
@@ -394,16 +449,55 @@ func (c *ServerCommand) Run(args []string) int {
 				return true
 			}
 
-			if err := sd.RunServiceDiscovery(c.WaitGroup, c.ShutdownCh, coreConfig.AdvertiseAddr, activeFunc, sealedFunc); err != nil {
+			if err := sd.RunServiceDiscovery(c.WaitGroup, c.ShutdownCh, coreConfig.RedirectAddr, activeFunc, sealedFunc); err != nil {
 				c.Ui.Error(fmt.Sprintf("Error initializing service discovery: %v", err))
 				return 1
 			}
 		}
 	}
 
+	handler := vaulthttp.Handler(core)
+
+	// This needs to happen before we first unseal, so before we trigger dev
+	// mode if it's set
+	core.SetClusterListenerSetupFunc(vault.WrapListenersForClustering(clusterAddrs, handler, c.logger))
```

Comment:
This doesn't even exist in current code :-)

On Aug 18, 2016 19:08, "Vishal Nayak" <notifications@github.com> wrote:

> In command/server.go
> <https://github.com/hashicorp/vault/pull/1721#discussion_r75405287>:
>
> >  				c.Ui.Error(fmt.Sprintf("Error initializing service discovery: %v", err))
> >  				return 1
> >  			}
> >  		}
> >  	}
> >
> > +	handler := vaulthttp.Handler(core)
> > +
> > +	// This needs to happen before we first unseal, so before we trigger dev
> > +	// mode if it's set
> > +	core.SetClusterListenerSetupFunc(vault.WrapListenersForClustering(clusterAddrs, handler, c.logger))
>
> c.clusterListenerSetupFunc will error out eventually. But...
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/hashicorp/vault/pull/1721/files/d497de45b3144200a4be257d228ef1a4f1d64ce2#r75405287>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AABv03-3W5_ulVL8vWsbo2Xb6ThPIPjdks5qhOX2gaJpZM4JieOG>
> .
>